etl/assets.py

```diff
@@ -1,71 +1,96 @@
 import json
-import os
-from pathlib import Path
 
-from dagster import asset, define_asset_job
-
-from etl.resources import (
-    OpenAiEmbeddingModelResource,
-    OpenAiGenerativeModelResource,
-    WikipediaReaderResource,
+from dagster import asset
+from langchain_core.vectorstores import VectorStore
+
+from etl.embedding_model_pipelines import OpenAiEmbeddingModelPipeline
+from etl.generative_model_pipelines import OpenAiGenerativeModelPipeline
+from etl.models import (
+    DocumentTuple,
+    OpenAiPipelineConfig,
+    OpenAiSettings,
+    RecordTuple,
+    data_files_config_from_env_vars,
+    output_config_from_env_vars,
 )
+from etl.readers import WikipediaReader
+from etl.utils import create_documents
 
 
 @asset
-def wikipedia_articles_from_storage(
-    wikipedia_reader_resource: WikipediaReaderResource,
-):
+def wikipedia_articles_from_storage() -> RecordTuple:
     """Materialize an asset of Wikipedia articles."""
 
-    return tuple(wikipedia_reader_resource.read())
+    return RecordTuple(
+        records=tuple(
+            WikipediaReader(data_files_config=data_files_config_from_env_vars).read()
```

Comment:
This code is in an `asset` so you should declare a dependency on `data_files_config` instead of importing a singleton. If you need the singleton for code that's not reachable from an asset then I'd still re-export the singleton as a resources dependency in your `Definitions`.

Reply:
Yes, I have removed all singletons and replaced them with resources.